pymeasure/instruments/thyracont/transmitter.py

```diff
@@ -0,0 +1,491 @@
+from pyvisa.constants import Parity, StopBits
+
+import pyvisa as pv
+from pyvisa import VisaIOError  # noqa: F401 for dependencies
+
+
+def calculateChecksum(message):
+    """Calculate the checksum for string `message`."""
+    value = 0
+    for i in range(len(message)):
+        value += ord(message[i])
+    return chr(value % 64 + 64)
+
+
+def compose_data(value):
+    """Generate a string with the length of `value` and the `value` itself afterwards.
+
+    :param value: Value to send to the device.
+    """
+    value = f"{value}"
+    return f"{len(value):02}{value}"
+
+
+source = {
+    'combination': 0,
+    'pirani': 1,
+    'piezo': 2,
+    'hotCathode': 3,
+    'coldCathode': 4,
+    'ambient': 6,
+    'relative': 7,
+}
+
+
+class Sources(IntEnum):
+    COMBINATION = 0
+    PIRANI = 1
+    PIEZO = 2
+    HOT_CATHODE = 3
+    COLD_CATHODE = 4
+    AMBIENT = 6
+    RELATIVE = 7
+
+
+gas_factor = Channel.control(
+    "0C{ch}00", "0C{ch}%s", "Control the gas correction factor.",
+    values=(0.2, 8),
+    validator=validators.strict_range,
+    set_process=compose_data,
+)
+
+
+class SensorChannel(Channel):
+    """Generic channel for individual pressure sensors of a Transmitter."""
+    _id = -1  # obligatory channel number, define in channel types
+
+    def __init__(self, parent, id):
+        if id is None:
+            id = self._id
+        assert id == self._id, f"Pirani ID has to be {self._id}."
+        super().__init__(parent, id=self._id)
+
+    pressure = Channel.measurement(
+        "0M{ch}00", """Get the current pressure in mbar.""",
+        preprocess_reply=lambda r: r.replace("UR", "0").replace("OR", "inf"),
+    )
+
+
+class Pirani(SensorChannel):
+    """Pirani sensor channel.
+
+    A Pirani sensor measures the heat transfer in the gas. The reading depends on the gas type.
+    The sensor reading is linear to the real pressure below some threshold, around 1 mbar.
+    You may control a gas factor with :attr:`gas_factor`.
+    """
+
+    _id = 1
+
+    gas_factor = gas_factor
+
+    statistics = Channel.measurement(
+        "0PM011",
+        """Get the sensor statistics: wear in percent (negative: corrosion,
+            positive: contamination), time since last adjustment in hours.""",
+        separator="A",
+        cast=int,
+        get_process=lambda vals: [v / 4 for v in vals],
+    )
+
+
+class Piezo(SensorChannel):
+    """Piezo sensor channel. A piezo sensor is independent of the gas present."""
+
+    _id = 2
+
+
+class HotCathode(SensorChannel):
+    """Hot cathode sensor channel."""
+
+    _id = 3
+
+    filament_mode = Instrument.control(
+        "0FC00", "2FC01%i", """The hot cathode filament control setting.""",
+        values={"2 if 1 defect": 0,
+                "Filament1": 1,
+                "Filament2": 2,
+                "toggle>1mbar": 3},
+        validator=validators.strict_discrete_set,
+        map_values=True,
+        cast=int,
+        check_set_errors=True,
+    )
+
```

Comment:
```suggestion

    degas = Instrument.control(
        "0DG00", "2DG01%i",
        """Control the degas mode.""",
        values={True: 1, False: 0},
        map_values=True,
        validator=validators.strict_discrete_set,
        check_set_errors=True,
    )

    enabled = Instrument.control(
        "0CC00", "2CC01%i",
        """Control the state of the cathode.""",
        values={True: 1, False: 0},
        map_values=True,
        validator=validators.strict_discrete_set,
        check_set_errors=True,
    )
```

Reply:
I changed it to 'sensor_enabled' in order to comply with pymeasure style.